drop-transfer/src/check.rs

```diff
@@ -0,0 +1,129 @@
+use std::{net::SocketAddr, ops::ControlFlow, sync::Arc};
+
+use hyper::StatusCode;
+use slog::{debug, info, warn, Logger};
+use tokio_util::sync::CancellationToken;
+
+use crate::{auth, protocol, service::State, tasks::AliveGuard, IncomingTransfer, Transfer};
+
+pub(crate) fn spawn(
+    state: Arc<State>,
+    xfer: Arc<IncomingTransfer>,
+    logger: Logger,
+    guard: AliveGuard,
+    stop: CancellationToken,
+) {
+    let id = xfer.id();
+    let job = run(state, xfer, logger.clone());
+
+    tokio::spawn(async move {
+        let _guard = guard;
+
+        tokio::select! {
+            biased;
+
+            _ = stop.cancelled() => {
+                debug!(logger, "Check job stop for transfer: {id}");
+            },
+            _ = job => (),
+        }
+    });
+}
+
+async fn run(state: Arc<State>, xfer: Arc<IncomingTransfer>, logger: Logger) {
+    loop {
+        tokio::time::sleep(drop_config::ALIVE_CHECK_INTERVAL).await;
+
+        if !state.transfer_manager.is_incoming_alive(xfer.id()).await {
+            break;
+        }
+
+        if make_request(&state.auth, &xfer, &logger).await.is_break() {
+            break;
+        }
+    }
```

Comment:
Could you explain this flow a bit? Why `make_request` is repeatedly called in the loop?

Reply:
`make_request()` ping the sender to check if the transfer is alive. According to Slack discussion on the problem, the agreed-upon solution was to ping periodically the sender, hence the loop

Reply:
I understand that the solution is to periodically ping the peer with `GET /check` to see if the transfer is still alive but am wondering about the code readability. From my perspective, I would expect:

```
make_request().await
loop {
  if !state.transfer_manager.check_if_alive(xfer.id()).await {
     ...
}
```